Restore dropped support for Innolux panel

pelwell opened the pull request:
In the move from 5.15 to 6.0 & 6.1, a commit adding support for an Innolux LCD panel (the catchily-named AT056tN53V1) was lost. Restore that support in a separate commit.

See https://forums.raspberrypi.com/viewtopic.php?p=2063287#p2063287.

aBUGSworstnightmare-rpi committed:
drm/panel: simple: Add Innolux AT056tN53V1 5.6" VGA

Add support for the Innolux AT056tN53V1 5.6" VGA (640x480) TFT LCD
panel.

Signed-off-by: Joerg Quinten <aBUGSworstnightmare@gmail.com>

pelwell commented:
@aBUGSworstnightmare-rpi Does this look OK to you?

6by9 commented:
It looks like we've got the panel patch, adding MEDIA_BUS_FMT_BGR666_1X24_CPADHI, and dt-binding merged into one. C'est la vie. 

Just be aware that official support for MEDIA_BUS_FMT_BGR666_1X24_CPADHI is coming in 6.2 via https://github.com/torvalds/linux/commit/2468e0195c09706896712a29030132e7f1bd0bcc, so this will have a conflict.

pelwell commented:
Noted re 6.2. How many commits would you like this in?

6by9 commented:
> How many commits would you like this in?

I'm not fussed over splitting it, just noting that we're potentially causing ourselves grief in forward porting.

aBUGSworstnightmare-rpi commented:
> It looks like we've got the panel patch, adding MEDIA_BUS_FMT_BGR666_1X24_CPADHI, and dt-binding merged into one. C'est la vie.
> 
> Just be aware that official support for MEDIA_BUS_FMT_BGR666_1X24_CPADHI is coming in 6.2 via [torvalds/linux@2468e01](https://github.com/torvalds/linux/commit/2468e0195c09706896712a29030132e7f1bd0bcc), so this will have a conflict.

Sorry, but MEDIA_BUS_FMT_BGR666_1X24_CPADHI is available in 6.1 as well https://github.com/raspberrypi/linux/blob/3fff7f771b67562955c02775e1ca472e244eab61/include/uapi/linux/media-bus-format.h#L52 . bus_format no got changed (compared to what is was when submitting the pull-request).

In case you want to add the timing to panel simple you may want to change the clock to below (but can do that with a pull-request later); gives a 60Hz output as the panel is running a little out of spec atm.
	`.pixelclock = { 25200000, 25200000, 25200000}`

pelwell commented:
Am I to assume, because you didn't mention my PR, that it is at least correct (if not complete)?

pelwell committed:
SQUASH: Correct subdev-formats documentation

As well as needing the format numbers updating to match the upstream
additions, the documentation for one of the added formats was using
the wrong name.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>

pelwell commented:
> Am I to assume, because you didn't mention my PR, that it is at least correct (if not complete)?

It appears not. The new subdev format numbers, included upstream but not documented, don't match the documentation this PR adds. I've added a follow-up patch, to be squashed on merge, that uses the new numbers (and fixes a likely cut-and-paste error in the original commit).
